### ISSUE TITLE
v1.0.0 release :: Issue #1 :: Template (reactivity?)

### ISSUE THREAD

**Issue by kyooriouskoala** [TheAncientGoat/autoform-slingshot#10]
I have a page that has two sections:

**Top section:**
Form to add new product

**Bottom section:**
Recently added products + (hidden) form

On the recently added product section, each of the product item in the list has an edit button and upon clicking it, the layout of that particular item will expand to show the form.

When editing the image upload field (removing existing image and choosing a new one), the blob image that appears on the "edit current product" form also appears on the "add new product" form which is not supposed to happen.

**Comment by timbrandin** [comment]
Did you say you had multiple forms with the same field-name? Because I can imagine that being a problem, we'd just need to add the collection to the template to solve that. Or we could add field with the template instance id in the SlingshotAutoformFileCache to separate the two.

But I'd rather go for the collection as a template instance variable, we'd just need to make sure we can get access to it everywhere.

**Comment by kyooriouskoala** [comment]
I have one single `method` form on the top section of the page and multiple `method-update` forms on the bottom section.

I am aware that it can be problematic to have multiple forms as it creates confusion with the data that is being saved. 

Fortunately since I design it so that the user can only edit one item at a time by hiding the rest of the `method-update` field, this apparently solves that issue. 

Also, it doesn't actually affect the `method` form at all regardless of whether I'm only showing one or all and when editing the `method-update` form. None of the `method-update` form field value appears on `method` form field. So ... I'm kind of surprised to see the blob image appears on the `method` form.

**Comment by timbrandin** [comment]
I think we can solve this pretty ease, with the solution as I mentioned
above keeping track of the cache per template right?

2015-08-03 13:46 GMT+02:00 kyooriouskoala notifications@github.com:

> I have one single method form on the top section of the page and multiple
> method-update forms on the bottom section.
> 
> I am aware that it can be problematic to have multiple forms as it creates
> confusion with the data that is being saved.
> 
> Fortunately since I design it so that the user can only edit one item at a
> time by hiding the rest of the method-update field, this apparently
> solves that issue.
> 
> Also, it doesn't actually affect the method form at all regardless of
> whether I'm only showing one or all and when editing the method-update
> form. None of the method-update form field value appears on method form
> field. So ... I'm kind of surprised to see the blob image appears on the
> method form.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/TheAncientGoat/autoform-slingshot/issues/10#issuecomment-127206004
> .

## 

Cheers,
// Tim Brandin @timbrandin https://twitter.com/#!/timbrandin
http://timbrandin.com
http://studiointeract.com

**Comment by timbrandin** [comment]
@kyooriouskoala I've added a PR for all your issues, could you have a look?

**Comment by kyooriouskoala** [comment]
Thanks, I'll check it out soon!

**Comment by kyooriouskoala** [comment]
Hi @timbrandin, was just about to update `timbrandin:autoform-slingshot` and noticed that it is still at 1.0.0. 

Could you publish a new version with the fixes like what you did the first time, please? Pardon the inefficiency, I'm new to git; designer learning how to code here. Thanks!

**Comment by ryanswrt** [comment]
I'll merge and deploy it soon

**Comment by kyooriouskoala** [comment]
Thanks, @TheAncientGoat. Just to avoid misunderstanding, I have not tested @timbrandin's latest PR since I am not familiar with git. 

The way I tested his very first PR for this package was by using his own published package `timbrandin:autoform-slingshot`; hence my request to update his package so that I can immediately help to test.

**Comment by timbrandin** [comment]
Ok, its published 1.1.0, I'll bump the version in this branch too.

It would be super nice if we could get you into testing on Git and the and
PRs themselves:
Here's a nice guide to set it up:
https://help.github.com/articles/checking-out-pull-requests-locally/

2015-08-04 8:51 GMT+02:00 kyooriouskoala notifications@github.com:

> Thanks, @TheAncientGoat https://github.com/TheAncientGoat. Just to
> avoid misunderstanding, I have not tested @timbrandin
> https://github.com/timbrandin's latest PR since I am not familiar with
> git.
> 
> The way I tested his very first PR for this package was by using his own
> published package timbrandin:autoform-slingshot; hence my request to
> update his package so that I can immediately help to test.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/TheAncientGoat/autoform-slingshot/issues/10#issuecomment-127499792
> .

## 

Cheers,
// Tim Brandin @timbrandin https://twitter.com/#!/timbrandin
http://timbrandin.com
http://studiointeract.com

**Comment by kyooriouskoala** [comment]
Yes, I'd definitely want to be able to do that.  Been reading Git tutes and "practising" :)

Thank you! I will read that guide. In the mean time, I'll check out the published v1.1.0.

**Comment by timbrandin** [comment]
:+1:

2015-08-04 11:10 GMT+02:00 kyooriouskoala notifications@github.com:

> Yes, I'd definitely want to be able to do that. Been reading Git tutes and
> "practising" :)
> 
> Thank you! I will read that guide. In the mean time, I'll check out the
> published v1.1.0.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/TheAncientGoat/autoform-slingshot/issues/10#issuecomment-127535015
> .

## 

Cheers,
// Tim Brandin @timbrandin https://twitter.com/#!/timbrandin
http://timbrandin.com
http://studiointeract.com

**Comment by kyooriouskoala** [comment]
`timbrandin:autoform-slingshot@1.1.0`

Blob image still appears on the 'add product' form while editing the image of an existing product entry using the 'edit product' form.

**Comment by kyooriouskoala** [comment]
I managed to git clone repo and added the package locally! :D

**Comment by timbrandin** [comment]
Yeay, we're levelling! Next task is to find the problem by debugging the
code ;)

I did not have time to look at the problem yesterday, was trying to figure
out how to reproduce it with test, I should've created an app like yours
instead, that would've been much more time efficient. Though I'd really
love to have test for situations like these so we're sure we cover them.

2015-08-05 6:38 GMT+02:00 kyooriouskoala notifications@github.com:

> I managed to git clone repo and added the package locally! :D
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/TheAncientGoat/autoform-slingshot/issues/10#issuecomment-127843635
> .

## 

Cheers,
// Tim Brandin @timbrandin https://twitter.com/#!/timbrandin
http://timbrandin.com
http://studiointeract.com

**Comment by kyooriouskoala** [comment]
Hi @timbrandin, 

If you have yet to create a similar app for testing things out, I've uploaded a test app here: https://github.com/kyooriouskoala/proentry

**Comment by timbrandin** [comment]
Perfect, thx

Skickat från min iPhone
